File: src/main/org/robocode/BotCatable.scala
package org.robocode
import org.robocode.genenticalgorithm.Chromosome

trait BotCatable {
	def setMovement(action:RobotAction);
	def getTanks():java.util.List[Tank]
	def getChromosome():Chromosome
	def getGunHeading():Double
	def getHeading():Double
	def getEnergy():Double
	def getX():Double
	def getY():Double
	def getBattleFieldWidth():Double
	def getBattleFieldHeight():Double
}

File: src/main/org/robocode/BotCatable.java
package org.robocode;

import java.util.List;

import org.robocode.genenticalgorithm.Chromosome;

import robocode.BulletHitEvent;
import robocode.BulletMissedEvent;

public interface BotCatable {
	void setMovement(RobotActionType action);
	List<Tank> getTanks();
	
	Chromosome getChromosome();
	double getGunHeading();
	double getHeading();
	double getEnergy();
	double getX();
	double getY();
	double getBattleFieldWidth();
	double getBattleFieldHeight();
	long getTime();
	List<BulletHitEvent> getRecentBulletHitEvents();
	List<BulletMissedEvent> getRecentBulletMissedEvents();
}
